backend/corpora/dataset_processing/process.py

```diff
@@ -176,14 +177,16 @@ def create_artifact(
         ExtraArgs={"ACL": "bucket-owner-full-control"},
     )
 
-    DatasetAsset.create(
+    artifact = DatasetAsset.create(
         dataset_id=dataset_id,
         filename=file_base,
         filetype=artifact_type,
         type_enum=DatasetArtifactType.REMIX,
         user_submitted=True,
         s3_uri=join("s3://", artifact_bucket, bucket_prefix, file_base),
     )
+    if not artifact:
+        raise CorporaException
```

Comment:
Check if the dataset is tombstoned at the top of this function. If it's tombstoned, return. Otherwise, we now need to cleanup the file uploaded to s3 on line 173. It would be even better if we could check for a tombstone before we even start converting.

Reply:
if we check before we start converting we still run the risk of the user cancelling the download while the conversion is running

Reply:
we should check before each conversion. Ideally, I'd like to have a thread running that periodically checks, similar to how the download code work, but in lieu of that, we can add lot of check points to see if it's canceled.

Reply:
I feel like a thread with periodic checks will create more race conditions than just a check a creation

Reply:
I disagree about it adding more race condition since that was handled by having a separate tombstone column this only updated by the user. However, it's more overhead to develop and adds complexity. Checking before creation is sufficient.